### PR TITLE
feat(discovery): check node format

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/upstream.lua [hunk 1]
```diff
@@ -235,7 +235,8 @@ function _M.set_by_route(route, api_ctx)
 
         local dis = discovery[up_conf.discovery_type]
         if not dis then
-            return 500, "discovery " .. up_conf.discovery_type .. " is uninitialized"
+            local err = "discovery " .. up_conf.discovery_type .. " is uninitialized"
```
Comment (membphis) [inline apisix/upstream.lua:238]: Why do we need this style?

File: t/node/upstream-discovery.t [hunk 2]
```diff
@@ -331,3 +331,35 @@ qr/proxy request to \S+/
...
+            local uri = "http://127.0.0.1:" .. ngx.var.server_port .. "/hello"
+            local httpc = http.new()
+            local res, err = httpc:request_uri(uri, {method = "GET", keepalive = false})
+            ngx.say(res.status)
+        }
+    }
+--- response_body
+503
+--- error_log
+invalid nodes format:
```
Comment (membphis) [inline t/node/upstream-discovery.t:365]: we need to confirm the content of the error message

File: apisix/upstream.lua (hunk 1 above)
Comment (spacewander) [reply apisix/upstream.lua:238]: Avoid line too long

**Review by membphis — COMMENTED** [review]
LGTM

File: apisix/upstream.lua (hunk 1 above)
Comment (membphis) [reply apisix/upstream.lua:238]: The same length is more confusing. 

image

**Review by tokers — APPROVED** [review]

**Review by membphis — APPROVED** [review]

**Pull request merged**